nautobot_ssot/integrations/aci/jobs.py

```diff
@@ -71,19 +64,45 @@ def data_mappings(cls):
 
     def load_source_adapter(self):
         """Method to instantiate and load the ACI adapter into `self.source_adapter`."""
-        self.source_adapter = AciAdapter(job=self, sync=self.sync, client=aci_creds[self.apic])
+        if not self.device_site:
+            self.logger.info("Device Location is unspecified so will revert to specified Controller's Location.")
+        _sg = self.apic.external_integration.secrets_group
```

Comment:
This might fail and probably requires some additional checks :
- external integration is optional on controller
- secrets are optional on external integration

Reply:
I've added a custom Exception and checks for these potential errors with logs.